### ISSUE TITLE
PyQt5 uses `directory` instead of `dir` for members of QFileDialog

### ISSUE THREAD

**Issue by jasperges** [mottosso/Qt.py#321]
In all bindings you have the keyword argument `dir` for members of `QFileDialog`, but in PyQt5 it is `directory` (personally I think it's better because it doesn't redefine a built-in). For example `QFileDialog.getOpenFilename(dir="/home/me")` would be `QFileDialog.getOpenFileName(directory="/home/me")` in PyQt5.

I have no idea how difficult this would be, but personally I think the best solution would be that you can use either `dir` or `directory`. If that would be impractical, I would suggest `dir` because that is used by the majority of bindings.

**Comment by BigRoy** [comment]
@jasperges Are you sure? It seems to only use `directory` for `__init__` but the static methods still use `dir` according to this: https://doc.qt.io/qt-5/qfiledialog.html#getOpenFileName right?

Or is this old documentation?

**Comment by jasperges** [comment]
@jasperges Yes, I'm sure. :) See [here](https://www.riverbankcomputing.com/static/Docs/PyQt5/api/qtwidgets/qfiledialog.html#getOpenFileName).

I used to like the documentation of PyQt better then the one from PySide. But's that's now totally reversed.

**Comment by BigRoy** [comment]
@mottosso note that PySide2 and PyQt5 do it differenty as Jasper describes [here](https://github.com/getavalon/core/issues/408#issuecomment-517663346)

> As far as I can tell PyQt5 uses directory everywhere, where PySide2 (and also PyQt4) use dir.

What's the Qt.py way to fix this one? ;)

**Comment by mottosso** [comment]
Me and Jasper chatted about this prior to making an issue out of it, and decided it wasn't worth handling through Qt.py.

Initially we thought of making a QtCompat member.

```py
QtCompat.getOpenFilename()
```

But there are [quite a few](https://doc.qt.io/qt-5/qfiledialog.html#static-public-members) of these static members that are presumably all wrong (in PyQt) that we would need to add and I'd like to avoid having too much of anyone's code being calls to `QtCompat`; it should be rare.

The other option is "fixing" the original calls, like we've done by forwarding things from Qt 4's QtGui to QtWidgets for example. That would be the ideal option, as it would mean code you write with PySide also works in PyQt and vice versa. It just isn't clear how complex that would be, but a pull-request is welcome.

And for completeness, the "correct" one would be `dir`, simply because that's what PySide2 does, and Qt.py is all about making PySide2's interface work with all bindings. (It's also in the corresponding C++ call)

**Comment by MHendricks** [comment]
`Qt.QtCompat.QFileDialog` already provides a compatibility specifically for the PyQt4 static members `getOpenFileName`, `getOpenFileNames` and `getSaveFileName` return not matching the other bindings. It should be possible to implement a method for PyQt5 similar to the `_standardizeQFileDialog` decorator used in the PyQt4 bindings that adjusts the kwargs before passing them to PyQt5.
